shared/protocol/src/main/java/io/pravega/shared/protocol/netty/CommandEncoder.java

```diff
@@ -154,34 +203,71 @@ private void validateAppend(Append append, Session session) {
         if (append.isConditional()) {
             throw new IllegalArgumentException("Conditional appends should be written via a ConditionalAppend object.");
         }
-        Preconditions.checkState(bytesLeftInBlock == 0 || bytesLeftInBlock > TYPE_PLUS_LENGTH_SIZE,
-                "Bug in CommandEncoder.encode, block is too small.");
     }
 
-    private void breakFromAppend(ByteBuf data, ByteBuf pendingData, ByteBuf out) {
-        if (bytesLeftInBlock != 0) {
+    private boolean isChannelFree() {
+        return writerIdPerformingAppends == null;
+    }
+
+    private boolean isChannelOwner(UUID writerID, String segment) {
+        return writerID.equals(writerIdPerformingAppends) && segment.equals(segmentBeingAppendedTo);
+    }
+
+    private boolean startAppend(ChannelHandlerContext ctx, AppendBatchSizeTracker blockSizeSupplier, Append append, ByteBuf out) {
```

Comment:
Javadoc

Reply:
do you need javadoc method prototype description for this private method? please clarify

Reply:
As a rule no. But it provides a convenient place to explain what is going on in complex methods or ones that need to be used in a particular way. I just thought this method would benefit from one.

Reply:
Thanks @tkaitchuck 
 sure .. I will add the descriptions for the code changes.

Reply:
I have added the Javadoc for most of the methods in the file CommandEncoder class.